solidity/random-datasource/relaxed-commit/relaxedRandom.sol

```diff
@@ -1,76 +1,132 @@
+ *
+ *          The random datasource is currently only available on the
+ *          ethereum main-net & public test-nets (Ropsten, Rinkeby & Kovan).
+ *
+ */
+pragma solidity >= 0.5 < 0.6;
+
+import "github.com/oraclize/ethereum-api/oraclizeAPI_0.5.sol";
+
+contract RandomRelaxedExample is usingOraclize {
+
+    uint256 NUM_RANDOM_BYTES_REQUESTED = 7;
+    event LogNewOraclizeQuery(string _description);
+    event generatedRandomNumber(uint256 _randomUint);
+
+    constructor()
+        public
+    {
+        oraclize_setProof(proofType_Ledger);
+        update();
     }
 
-    // the callback function is called by Oraclize when the result is ready
-    // the oraclize_randomDS_proofVerify modifier prevents an invalid proof to execute this function code:
-    // the proof validity is fully verified on-chain
-    function __callback(bytes32 _queryId, string _result, bytes _proof)
+    function __callback(
+        bytes32 _queryId,
+        string memory _result,
+        bytes memory _proof
+    )
+        public
     {
-        if (msg.sender != oraclize_cbAddress()) throw;
-
-        if (oraclize_randomDS_proofVerify__returnCode(_queryId, _result, _proof) != 0) {
-            // the proof verification has failed, do we need to take any action here? (depends on the use case)
+        require(msg.sender == oraclize_cbAddress());
+        if (
+            oraclize_randomDS_proofVerify__returnCode(
+                _queryId,
+                _result,
+                _proof
+            ) != 0
+        ) {
+            /**
+             * @notice  The proof verification has failed! Handle this case
+             *          however you see fit.
+             */
         } else {
-            // the proof verification has passed
-            // now that we know that the random number was safely generated, let's use it..
-
-            newRandomNumber_bytes(bytes(_result)); // this is the resulting random number (bytes)
-
-            // for simplicity of use, let's also convert the random bytes to uint if we need
-            uint maxRange = 2**(8* 7); // this is the highest uint we want to get. It should never be greater than 2^(8*N), where N is the number of random bytes we had asked the datasource to return
-            uint randomNumber = uint(sha3(_result)) % maxRange; // this is an efficient way to get the uint out in the [0, maxRange] range
-
-            newRandomNumber_uint(randomNumber); // this is the resulting random number (uint)
+            /**
+             *
+             * @notice  The proof verifiction has passed!
+             *
+             *          Let's convert the random bytes received from the query
+             *          to a `uint256`. To do so, We define the variable
+             *          maxRange, where maxRange - 1 is the highest uint256 we
+             *          want to get. The variable maxRange should never be
+             *          greater than: 2 ^ (8 * NUM_RANDOM_BYTES_REQUESTED).
```

Comment:
Do you think could be useful to explain also why it must not be greater than the described value?

Reply:
Yes good idea. Rather than explain it though I decided to name the magic numbers as variables instead & rename `maxRange` which makes the whole thing much clearer without having to add yet more comments.